toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1753,6 +1898,56 @@ func createRawArtifact(workDirPath, devPath, name string) (err error) {
 	return shell.ExecuteLive(squashErrors, "dd", ddArgs...)
 }
 
+func createRDiffArtifact(workDirPath, devPath, rDiffBaseImage, name string) (err error) {
+	const (
+		signatureFileName = "./signature"
```

Comment:
Using a relative path here may lead to hard-to-debug errors, as it's relying on an assumption of what the current working directory is.

I'd recommend replacing this with a calculated absolute path.  e.g. if we expect the signtures file to live under `workDirPath`: `filepath.Join(workDirPath, "signature")`

Reply:
Following the same. line 1910 appends this and uses the workspace path.

---
In reply to: [573080788](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r573080788) [](ancestors = 573080788)

Reply:
Quick clarification: it looks like `signatureFileName` is passed as an argument to `rdiff` and `delta`. Do these programs automatically make this path relative to `rDiffBaseImage`? Or is there something I'm missing here.

Reply:
Yes. The versions of rdiff and delta I'm using was finicky about relative paths.

---
In reply to: [574707089](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r574707089) [](ancestors = 574707089)

Reply:
Good to know, thanks! In that case I'd recommend adding a quick comment explaining why its relative so in the future someone doesn't accidently try and make it an absolute path :).

The only other thing that I'd ask to be made an absolute path is the `os.Remove(signatureFileName)` on line  1920, since it's not reliant on `rdiff` and `delta` to automatically make the path based on `rDiffBaseIMage`.

Reply:
Fixed.

---
In reply to: [575389980](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r575389980) [](ancestors = 575389980)